runtime/domain.c

```diff
@@ -906,13 +906,46 @@ void caml_handle_gc_interrupt() {
   }
 }
 
+CAMLexport void caml_bt_leave_blocking_section_hook(void)
+{
+  dom_internal* self = domain_self;
+
```

Comment:
Can we put back the assert:
`Assert(caml_domain_alone() || self->backup_thread_running);`